controllers/logicalvolume_controller_test.go

```diff
@@ -0,0 +1,223 @@
+	})
+
+	setupResources := func(ctx context.Context, suffix string) topolvmv1.LogicalVolume {
+		node := corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node" + suffix,
+				Finalizers: []string{
+					topolvm.GetNodeFinalizer(),
+				},
+			},
+		}
+		err := k8sClient.Create(ctx, &node)
+		Expect(err).NotTo(HaveOccurred())
+
+		sc := storegev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "sc" + suffix,
+			},
+			Provisioner: topolvm.GetPluginName(),
+		}
+		err = k8sClient.Create(ctx, &sc)
+		Expect(err).NotTo(HaveOccurred())
+
+		ns := createNamespace()
+		pvc := corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pvc" + suffix,
+				Namespace: ns,
+				Annotations: map[string]string{
+					AnnSelectedNode: node.Name,
+				},
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				StorageClassName: &sc.Name,
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteOnce,
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: *resource.NewQuantity(1, resource.BinarySI),
+					},
+				},
+			},
+		}
+		err = k8sClient.Create(ctx, &pvc)
+		Expect(err).NotTo(HaveOccurred())
+
+		lv := topolvmv1.LogicalVolume{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "lv" + suffix,
+			},
+			Spec: topolvmv1.LogicalVolumeSpec{
+				NodeName: node.Name,
+			},
+		}
+		err = k8sClient.Create(ctx, &lv)
+		Expect(err).NotTo(HaveOccurred())
+
+		return lv
+	}
+
+	It("should add finalizer to LogicalVolume", func() {
+		startReconciler("-add-finalizer")
+
+		ctx := context.Background()
+
+		// Setup
+		lv := setupResources(ctx, "-add-finalizer")
+
+		// Verify
+		// ensure LV has finalizer
+		Eventually(func(g Gomega) bool {
+			err := k8sClient.Get(ctx, client.ObjectKeyFromObject(&lv), &lv)
+			if err != nil {
+				return false
+			}
+			return controllerutil.ContainsFinalizer(&lv, topolvm.GetLogicalVolumeFinalizer())
+		}).Should(Equal(true))
+	})
+
+	It("should not add finalizer to LogicalVolume when volume has pendingdeletion annotation", func() {
+		startReconciler("-pendingdeletion")
+
+		ctx := context.Background()
+
+		// Setup
+		lv := setupResources(ctx, "-pendingdeletion")
+
+		// ensure LV gets finalizer
+		Eventually(func(g Gomega) bool {
+			err := k8sClient.Get(ctx, client.ObjectKeyFromObject(&lv), &lv)
+			if err != nil {
+				return false
+			}
+			return controllerutil.ContainsFinalizer(&lv, topolvm.GetLogicalVolumeFinalizer())
+		}).Should(Equal(true))
+
+		// Add pending deletion key & remove finalizer
+		lv2 := lv.DeepCopy()
+		lv2.Annotations = map[string]string{
+			topolvm.GetLVPendingDeletionKey(): "true",
+		}
+		controllerutil.RemoveFinalizer(lv2, topolvm.GetLogicalVolumeFinalizer())
+
+		patch := client.MergeFrom(&lv)
+		err := k8sClient.Patch(ctx, lv2, patch)
+		Expect(err).NotTo(HaveOccurred())
+
+		// ensure LV finalizer is removed
+		Eventually(func(g Gomega) bool {
```

Comment:
Because the LogicalVolume reconciler may run after this check, I would like you to check that the finalizer is not added for 1 or 2 seconds.
I believe you can do it by using `Consistently` instead of `Eventually`.
cf. https://pkg.go.dev/github.com/onsi/gomega#Consistently

Reply:
Ah nice. That's done now